### ISSUE TITLE
Runners, torchbench, & the future

### ISSUE THREAD

**Issue by janeyx99** [pytorch/pytorch#143215]
The purpose of this issue is to centralize discussions regarding the state of our runners and torchbench,  in particular what should be expected when they go through transitions. It is a bit of a weird issue as this does not point to any codebase problems with pytorch/pytorch, but the intended discussion group spans beyond pytorch/benchmark so I'm putting the issue here.

### Context 
I recently wanted to send someone my optimizers unidash to discover that it was empty, oh no! I traced the problem back to the GHA in pytorch/benchmark that runs the benchmark (and writes the stats) to discover that it's been red again, for weeks, because the a100-runner runner it ran on had been removed: https://github.com/pytorch/benchmark/actions/workflows/userbenchmark-regression-detector.yml

I fixed it with @xuzhao9's help https://github.com/pytorch/benchmark/pull/2557 but two things picked at me:
(1) Not just my workflow, but all the userbenchmark workflows + anything else using A100 were not migrated, and were just failing: https://github.com/pytorch/benchmark/actions/workflows/userbenchmark-a100.yml and https://github.com/pytorch/benchmark/actions/workflows/v3-nightly.yml. Was I the first to notice because I'm the only one to care? Are there others who care but haven't been notified? We should migrate the workflows that do matter and delete the code for the ones that don't.
(2) It would have been nice to have a heads up. I believe there was a migration plan (cc @jeanschmidt @atalman @xuzhao9) as I saw some budding draft PRs, though I would have expected for all usages of the runner to be migrated before the runner was removed completely. Is this incident a fluke or should I shift expectations?

### Questions to answer

(1) Across the repos in the org (not just pytorch/pytorch), what is the expected transition plan for when a runner will no longer be running anymore? cc @seemethere 
(2) (or 1b) Does it matter whether it's a repo-level runner vs org-level runner?
(3) Do people care about the userbenchmarks in torchbench/are there owners (other than me)? If not, we should do a big code delete! cc @drisspg for torchao, @pytorch-dev-infra for nightly-V3.
(4) Is torchbench userbenchmarks still the recommended path for writing recurring benchmarks? Where are folks moving to? cc @xuzhao9 

### What I ultimately care about

That my optim benchmarks will have a home to keep running, that they won't surprisingly break on me if we could help it!

**Comment by huydhn** [comment]
So, I'm going over the list of runners we have at the moment to document which are there:

**On GCP**

1. [SHUTDOWN] Dev Infra GCP A100 runners, `gh-ci-gcp-a100`, were used by `pytorch/pytorch`.  They have been shut down a while ago to be replaced with `linux.aws.a100` runners on AWS.  `pytorch/benchmark` doesn't use them, so they can be completely removed now.  @jeanschmidt is on that.
2. [SHUTDOWN] TorchBench GCP A100 ARC Kubernetes cluster `a100-runner`.  This was used by `pytorch/benchmark`.  Its shutting down was the direct cause of this regression I think. This was a repo-only runners and was only available to `pytorch/benchmark`.
3. [IN USE] TritonBench GCP H100 ARC Kubernetes cluster `gcp-h100-runner`.  This is currently used by `pytorch-labs/tritonbench`, repo-only runners.

**On AWS**

1. [IN USE] Dev Infra AWS A100 runners `linux.aws.a100` runner.  This is available to all pytorch repos.

Some observations:

* Runners from Dev Infra are available for all pytorch repos.  We want to use them on TorchBench too.
* Runners from TritonBench are only available to that repo.
* I don't see many activities on TorchBench user benchmark to v3 bisect, but maybe this is because they are not properly maintained instead of the lack of interest there.
    * For bisecting part, I have chatted with @izaitsevfb on this before, we will need to see how we can build on top / refactoring this for future use.
    * For nightly-v3 and release user benchmark, release engineers are still including it as one of the final checks before each release, but its value is questionable.  IMO, they are not that useful in the current state and we need to see if we want to resurrect them (@atalman @kit1980)

**Comment by jeanschmidt** [comment]
Do we want to have the A100 and H100 instances in GCP that have been deployed in ad-hoc mode to be maintained? I believe we might want to reach out to all the authors to either migrate their workflows to the current instances or remove them.

Are there legitimate use cases for the existence of them except from the fact that we didn't offer those instances in the past?

**Comment by drisspg** [comment]
From one of your meta points @janeyx99 I think part of the problem is that the benchmarks we care about are far removed from the users. I know Jerry has started to look at how to author and maintain benchmarks for torchao within torchao.

**Comment by bhack** [comment]
Do you think we could have `L40S` coverage?

**Comment by atalman** [comment]
@huydhn we do still execute: https://github.com/pytorch/benchmark/blob/main/.github/workflows/userbenchmark-a100-release.yml for each release, I believe we where able to catch 2-3 regressions using these benchmarks during last two years that we run them. So we keep them alive. I believe we should still run these but extend. I believe there is an issue for this work https://github.com/pytorch/benchmark/issues/2468, @juliagmt-google looked into this.

**Comment by malfet** [comment]
> Do you think we could have `L40S` coverage?

We could (if you are referring to aws g6e.xlarge runners), question is what runner family they should replace)

I wonder if this should be discussed as part of the same issue, or someplace else but https://github.com/pytorch/test-infra/blob/main/.github/scale-config.yml has 30+ runners defined, perhaps good first step would be to add comments with their HW specs and what they are used for

**Comment by huydhn** [comment]
> @huydhn we do still execute: https://github.com/pytorch/benchmark/blob/main/.github/workflows/userbenchmark-a100-release.yml for each release

Got it, if that's the case, here is the fix to make it use AWS A100 runner https://github.com/pytorch/benchmark/pull/2560

**Comment by bhack** [comment]
> We could (if you are referring to aws g6e.xlarge runners), question is what runner family they should replace)

Yes I found some specific autotuning issues on this platform at:
https://github.com/pytorch/pytorch/pull/143098#issuecomment-2545805928

**Comment by soulitzer** [comment]
From triage review:
- Nikita: takeaway - define a process, what are the steps infra team should take when adding/removing runner.

**Comment by malfet** [comment]
> > We could (if you are referring to aws g6e.xlarge runners), question is what runner family they should replace)
> 
> Yes I found some specific autotuning issues on this platform at: [#143098 (comment)](https://github.com/pytorch/pytorch/pull/143098#issuecomment-2545805928)

Good point. Do you mind creating a specific issue for it? If PT compiler community cares for those types of issues, then runners can be added.

**Comment by bhack** [comment]
> Good point. Do you mind creating a specific issue for it? If PT compiler team / community cares for those types of issues, then runners can be added.

I've not opened a new ticket as we had historically this issue on different HWs. See https://github.com/search?q=repo%3Apytorch%2Fpytorch+Hardware+limit%3A101376&type=issues

So I don't know how we could really improve the coverage of these auto-tuning problems.

**Comment by atalman** [comment]
Fixed torchbench userbenchmark on a100: https://github.com/pytorch/benchmark/pull/2571 and release-test

**Comment by huydhn** [comment]
> Fixed torchbench userbenchmark on a100: [pytorch/benchmark#2571](https://github.com/pytorch/benchmark/pull/2571) and release-test

The workflow is running but I don't see the results on https://hud.pytorch.org/torchbench/userbenchmark yet for release-test.  So, there might still be an issue, maybe with the upload part.
